src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -318,8 +320,6 @@ public void move(Vector spdVec, double rot) {
    * @param rateLimit whether or not to use slew rate limiting
    */
   private void move(Vector spdVec, double rot, boolean rateLimit) {
-    spdVec = spdVec.copy();
```

Comment:
I believe you shouldn't modify the buffer given to you. You should make a private buffer in the class called `moveSpd`

Reply:
what do you mean by a buffer?

Reply:
A buffer is a region of memory that is used to store some things while it is being moved around. A `Vector` is in this necessarily is a buffer as that's what it does. I don't think your drive algorithm should modify the input given to it. You should instead allocate another vector to copy the input into so that you can mutate that instead. This kind of modification can lead to bizarre bugs if you make assumptions about the state of the input vector after the call...

Reply:
I know, but I checked carefully, and the whole point of this branch is to save memory by not allocating as many vectors

Reply:
> I know, but I checked carefully, and the whole point of this branch is to save memory by not allocating as many vectors

Allocating one more vector at construction isn't a high cost. I think you should do it as to not mutate the input vector.

Reply:
sounds good